pkg/lib/serverTime.js

```diff
@@ -321,13 +320,131 @@ export function ServerTime() {
                     console.warn("failed to load time servers", error);
                     return result;
                 });
-    };
+    }
+
+    function set_custom_ntp_timesyncd(servers, enabled) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
 
-    self.set_custom_ntp = function (servers, enabled) {
         const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function get_custom_ntp_chronyd() {
+        const result = {
+            supported: true,
+        };
+
+        const enabled_file = cockpit.file("/etc/chrony-cockpit-sources/custom.sources", { superuser: "try" });
+        const disabled_file = cockpit.file("/etc/chrony-cockpit-sources/custom.disabled", { superuser: "try" });
+
+        function parse_servers(data) {
+            if (!data)
+                return [];
+            const servers = [];
+            data.split("\n").forEach(function(line) {
+                const parts = line.split(" ");
+                if (parts[0] == "server")
+                    servers.push(parts[1]);
+            });
+            return servers;
+        }
+
+        return enabled_file.read()
+                .then(data => {
+                    if (data) {
+                        result.servers = parse_servers(data);
+                        result.enabled = true;
+                        return Promise.resolve(result);
```

Comment:
Why not simply `return result`? You use this pattern a few times here, but the last item in a promise chain can/should just be the eventually computed value.

Reply:
Yes, agreed.